### PR TITLE
🐛 fix flaky test --> add minimum wait time that is large enough

### PR DESCRIPTION
**Pull request by mrnicegyu11**
This test would always run on my local machine, but fail sometimes on github actions. A timeout is needed here by design since a directory watcher is tested.

By decreasing the minimum timeout period successively I could make the test flaky on my local PC.
Arbitrarily we now set the minimum wait time to 3 seconds, which is 1 second more than the minimum 2sec waittime reported in failed test on github actions 

- Fixes https://github.com/ITISFoundation/osparc-simcore/issues/3694

Bonus:
Fixes (maybe impactful?) typo `sire_effects` --> `side_effects`

### CODE REVIEW

**Comment by mrnicegyu11** [comment]
Adresses https://github.com/ITISFoundation/osparc-simcore/issues/3694

**Review by pcrespov — APPROVED** [review]

**Review by pcrespov — COMMENTED** [review]
thx! Greatly appreciated! :-)

**Review by matusdrobuliak66 — APPROVED** [review]
Thanks

File: services/dynamic-sidecar/tests/unit/test_modules_outputs_watcher.py [hunk 1]
```diff
@@ -359,9 +359,11 @@ async def test_port_key_sequential_event_generation(
             )
         )
 
-    # Waiting for events o finish propagation and changes to be uploaded
+    # Waiting for events to finish propagation and changes to be uploaded
     MARGIN_FOR_ALL_EVENT_PROCESSORS_TO_TRIGGER = 1
-    sleep_for = max(wait_interval_for_port) + MARGIN_FOR_ALL_EVENT_PROCESSORS_TO_TRIGGER
+    sleep_for = max(
+        [max(wait_interval_for_port) + MARGIN_FOR_ALL_EVENT_PROCESSORS_TO_TRIGGER, 3]
```
Comment (GitHK) [inline services/dynamic-sidecar/tests/unit/test_modules_outputs_watcher.py:365]:
```suggestion
        max(wait_interval_for_port) + MARGIN_FOR_ALL_EVENT_PROCESSORS_TO_TRIGGER, 3
```

Comment (GitHK) [reply services/dynamic-sidecar/tests/unit/test_modules_outputs_watcher.py:365]: Pleas give a name to the 3 constant something like `MINIMUM_TIME_TO_WAIT = 3`

**Review by GitHK — APPROVED** [review]
Nice, please consider my comments

**Review by sanderegg — APPROVED** [review]

**Comment by pcrespov** [comment]
> Reassigning to @pcrespov as requested.

@mrnicegyu11 :-))) I meant "re-request review" :-) sorry for the confusion!

**Pull request merged**